### PR TITLE
[SW-1231] WIP: Training API more similar to Spark

### PR DESCRIPTION
**Pull request by jakubhava**
Very very WIP, for Scala, probably not complete even for Scala.

Because the only input will be array of feaures ( doubles) we can simplify the code and throw out all the categorical conversions

### CODE REVIEW

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala [hunk 1]
```diff
@@ -85,12 +80,12 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag, M <: SparkModel[M]
...
     }
     water.DKV.put(trainFrame)
-    
+
     // Train
     val model: M with H2OModelParams = trainModel(getParams)
 
     // pass some parameters set on algo to model
-    model.setFeaturesCols($(featuresCols))
+    model.setFeaturesCols(Array($(featuresCol)))
```
Comment (jakubhava) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala]: For prediction time, still keep the api same, iterate in small steps :)

File: ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala [hunk 2]
```diff
@@ -47,18 +47,17 @@ abstract class H2OAlgorithm[P <: Model.Parameters : ClassTag, M <: SparkModel[M]
...
-      setFeaturesCols(dataset.columns.filter(n => n.compareToIgnoreCase(getLabelCol()) != 0))
+    if(dataset.columns.contains(getFeaturesCol())){
+      throw new IllegalArgumentException(s"Can not find features column '${getFeaturesCol()}' in the dataset")
     }
 
-    val cols = getFeaturesCols().map(col) ++ Array(col(getLabelCol()))
-    val input = hc.asH2OFrame(dataset.select(cols: _*).toDF())
+    // To find the model, we just need two columns, label & features
+    // We are sure that features column is array of doubles
+    val input = hc.asH2OFrame(dataset.select(getLabelCol(), getFeaturesCol()).toDF())
```
Comment (jakubhava) [inline ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAlgorithm.scala]: Actually, we will need to convert also fold column and weight column, if set as they need to be available on the H2O frame

**Comment by jakubhava** [comment]
Should go to next major release as it will unblock us with pipeline improvements

File: ml/src/main/scala/org/apache/spark/ml/h2o/param/H2OModelParams.scala [hunk 3]
```diff
@@ -16,73 +16,41 @@
...
+  setDefault(predictionCol -> "prediction")
   setDefault(convertUnknownCategoricalLevelsToNa -> false)
 
   //
   // Getters
   //
-  /** @group getParam */
-  @DeprecatedMethod("getLabelCol")
-  def getPredictionsCol(): String = getLabelCol()
+  def getPredictionCol(): String = $(predictionCol)
```
Comment (mn-mikke) [inline ml/src/main/scala/org/apache/spark/ml/h2o/param/H2OModelParams.scala]: remove

**Comment by jakubhava** [comment]
Closing this one due to number of conflicts, it will be safer to start this change over

**Pull request closed without merging**